libbeat/common/docker/client.go

```diff
@@ -34,6 +34,13 @@ import (
 func NewClient(host string, httpClient *http.Client, httpHeaders map[string]string) (*client.Client, error) {
 	log := logp.NewLogger("docker")
 
+	if host == "" {
+		host = os.Getenv("DOCKER_HOST")
+		if host == "" {
+			host = "unix:///var/run/docker.sock"
+		}
+	}
+
```

Comment:
Why is this needed here?

Reply:
So when you create the client without a host it will fallback to using the environment variable or if that is not set to default that docker expects.

Reply:
I think that this change wouldn't be needed for the change in tests, and this may affect beats functionality. For example I am worried that this can affect non-unix platforms, in Windows the default is different (`npipe:////./pipe/docker_engine`).

I agree that supporting `DOCKER_HOST` (or other environment variables) and having a sane default would be a good thing to do, but this may require further discussions, we could do it in a different PR.

Wdyt about using by now `client.DefaultDockerHost` in tests instead of `""`? (`client` as imported from `github.com/docker/docker/client`).

Reply:
Forgot about the Windows case.

I like the usage of `client.DefaultDockerHost` but how would the code path be able to pick up the environment variable setting, unless we want the caller to always be specific.

Another option would be to change the `unix:///var/run/docker.sock` to `client.DefaultDockerHost`. Then it would work for Unix and Windows. Thoughts?

Reply:
Yes, I would say to remove the changes in this file, and use `client.DefaultDockerHost` in tests instead of the empty string (not sure if this can affect Windows using a docker machine, we'll see).

In a future change we can consider adding support for `DOCKER_HOST` environment variable and the fallback to the default value in `NewClient()`.

Reply:
I will make the change as suggested, but do have a worry.

I only worry that removing the the reading of environment variable might break some tests runners. As the current tests use `client.NewEnvClient` that reads from the environment variables. Changing it to always use `client.DefaultDockerHost` would mean that the tests will not read the environment variable.

Reply:
> I only worry that removing the the reading of environment variable might break some tests runners. As the current tests use `client.NewEnvClient` that reads from the environment variables. Changing it to always use `client.DefaultDockerHost` would mean that the tests will not read the environment variable.

Let's try, I don't think we were expecting `DOCKER_HOST` to work.

Reply:
Done, give it another look.

Reply:
Thanks, let's wait for tests in CI.